### PR TITLE
DynamoDB Memory Driver

### PR DESCRIPTION
**Pull request by collindutter**
Working well in my testing, but still need to create unit tests.

### CODE REVIEW

**Comment by vasinov** [comment]
I changed the naming structure and namespace for memory drivers in the last update. Could you update this driver to match it? Thanks!

**Comment by collindutter** [comment]
@vasinov this one is good for review. Added tests and updated structure/namespace.

**Comment by collindutter** [comment]
Oops, the old driver is still committed. Will remove today.

**Comment by collindutter** [comment]
@vasinov tests are failing now (they were passing before), and it looks like a hugging face timeout?

File: griptape/drivers/memory/conversation/dynamodb_conversation_memory_driver.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from attr import define, field
+from boto3 import resource
+from griptape.drivers import BaseConversationMemoryDriver
+from griptape.memory.structure import ConversationMemory
+
+
+@define
+class DynamoDbConversationMemoryDriver(BaseConversationMemoryDriver):
+    aws_region: str = field(default=None, kw_only=True)
```
Comment (vasinov) [inline griptape/drivers/memory/conversation/dynamodb_conversation_memory_driver.py:9]: Let's make the default "us-east-1"

File: griptape/drivers/memory/conversation/dynamodb_conversation_memory_driver.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    table_name: str = field(default=None, kw_only=True)
+    partition_key: str = field(default=None, kw_only=True)
+    value_attribute_key: str = field(default=None, kw_only=True)
+    partition_key_value: str = field(default=None, kw_only=True)
```
Comment (vasinov) [inline griptape/drivers/memory/conversation/dynamodb_conversation_memory_driver.py:13]: Sounds like of those should be required, right? If so, we should remove the defaults.

File: griptape/drivers/memory/conversation/dynamodb_conversation_memory_driver.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        if "Item" in response and self.value_attribute_key in response["Item"]:
+            memory_value = response["Item"][self.value_attribute_key]
+
+            memory = ConversationMemory.from_json(memory_value)
+
+            memory.driver = self
+
+            return memory
+        return None
```
[the remark this answers is not in the document]
Comment (collindutter) [reply griptape/drivers/memory/conversation/dynamodb_conversation_memory_driver.py]: @vasinov we should probably define some pylint rules for the project. Fixed for now though.
image

[the remark this answers is not in the document]
Comment (vasinov) [reply griptape/drivers/memory/conversation/dynamodb_conversation_memory_driver.py]: Yep, for sure. Will add as an issue :)

**Pull request merged**